apps/10address-book-ajax/script.js

```diff
@@ -2,4 +2,53 @@
 
 $(document).ready(function() {
   // You code here
+
+  var settings = {
+    success: successCallback
+  }
+
+  function successCallback(users){
+    users.forEach(handleUser)
+  }
+
+  function handleUser(users){
+    var str = "<tr>" +
+    "<td>" + users.id + "</td>" +
+    "<td>" + users.first_name + "</td>"+
+    "<td>" + users.last_name + "</td>"+
+    "<td> <a href='#' data-id=" + users.id+ ">view</a>" + "<td>" +
+    "</tr>"
+
+    $('tbody').append(str);
+  }
+
+  $.ajax('https://reqres-api.herokuapp.com/api/users', settings);
+
+  // In a success callback, pass in users as your reponse, and the iterate --aka loop-- over each user using JavaScript forEach.
+
+  // In each loop, create a var called str that builds an html string that matches the <tr></tr> in the html markup, but with the user keys. At the end of each loop, append the str to the tbody element.
+
+  $('body').on('click','a',function(users) {
```

Comment:
You need to pass in `event`, not "users" to perform an `event.preventDefault();`

Everything else looks great!